### PR TITLE
use some pattern matching

### PR DESCRIPTION
**Pull request by SimonCropp**

### CODE REVIEW

**Comment by hishamco** [comment]
This introduced a lot of changes which of course will introduce a merge conflict in the pending PRs, I prefer to hold this for now

**Comment by SimonCropp** [comment]
@hishamco seems a reason to block any kind of pr that cleans up code

**Comment by hishamco** [comment]
I understand

Screenshot 2024-02-04 141143

This is a sample for some of my PRs after introducing a github bot :)

**Comment by mariojsnunes** [comment]
Was there a discussion about updating code to use more pattern matching?
One could argue it's not worth it, as it becomes too different from common coding languages.
Maybe I'm just not used to it... I'm genuinely curious to know how others feel about this...

File: src/OrchardCore.Modules/OrchardCore.AdminDashboard/Views/DashboardWidget-ContentsMetadata.DetailAdmin.cshtml [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 }
 
-@if (dashboardFeature != null && dashboardFeature.IsManageRequest)
+@if (dashboardFeature is {IsManageRequest: true})
```
Comment (MikeAlhayek) [inline src/OrchardCore.Modules/OrchardCore.AdminDashboard/Views/DashboardWidget-ContentsMetadata.DetailAdmin.cshtml:12]: I don't like this. it makes the code harder to read.

**Comment by MikeAlhayek** [comment]
I do not like all the changes here. Some of these changes will make the code not easily readable.

**Pull request closed without merging**